### PR TITLE
Most of 7.22b (KV Files)

### PR DESCRIPTION
**Pull request by POCKETMONEY99**
Most of the  Key value changes for 7.22. I also added in most of the new KV files for the abilities. 
List of Things I haven't done:
-Monkey King Aghs, (Custom File)
-Dragon Knight Aghs (custom File)
-Consumable aghs (It has the same name as level 2 aghanims.)

Some files also have KV which havent been cleaned up by Vavle.
- Anti-mage Mana Break 
-Beastmaster Inner Beast
-Ember Spirit Searing Chains
-Ember Spirit Flame Guard
-Monkey King Wukongs Command (Cooldown reduction)
-Pangolier Swashbuckle

Lycan is the only hero who needs a new aghs upgrade. His aghs won't do anything in OAA as it requires creep waves to work.

### CODE REVIEW

File: game/scripts/npc/items/item_spirit_vessel.txt [hunk 1]
```diff
@@ -117,12 +117,12 @@
       "13"
       {
         "var_type"        "FIELD_INTEGER"
-        "hp_regen_reduction_enemy"  "70"
+        "hp_regen_reduction_enemy"  "60"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/items/item_spirit_vessel.txt]: We should make this scale: `60 70 80 90 100`

**Comment by POCKETMONEY99** [comment]
> 
> 
> Do you wanna finish this by yourself? I can help if you want. I can do the stuff that's left in a separate PR.

I will finish this

**Comment by POCKETMONEY99** [comment]
Travis is saying that windrun shouldn't have a special value for "scepter_movespeed_bonus_pct_tooltip". Apart from this everying this else works. 
"09"
			{
				"var_type"				"FIELD_INTEGER"
				"scepter_movespeed_bonus_pct_tooltip"	"105"

File: game/scripts/npc/items/item_skadi.txt [hunk 2]
```diff
@@ -69,22 +69,17 @@
       "04"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "cold_movement_speed"                             "-35 -37 -39 -41 -43"
+        "cold_slow_melee"                                 "-20 23 26 29 31"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/items/item_skadi.txt]: It needs to be negative values for the slow.

File: game/scripts/npc/abilities/bloodseeker_rupture.txt [hunk 3]
```diff
@@ -49,33 +49,33 @@
...
-        "duration"                                        "8 10 12 14 16"
+        "var_type"          "FIELD_FLOAT"
+        "duration"          "12"
       }
       "02"
       {
-        "var_type"                                        "FIELD_INTEGER"
-        "movement_damage_pct"                             "8"
+        "var_type"          "FIELD_INTEGER"
+        "movement_damage_pct"   "30 45 60 90 120"
```
[the remark this answers is not in the document]
Comment (POCKETMONEY99) [reply game/scripts/npc/abilities/bloodseeker_rupture.txt]: ok

File: game/scripts/npc/abilities/chen_divine_favor.txt [hunk 4]
```diff
@@ -7,55 +7,46 @@
...
-		"ID"                    "7306"														// unique ID number for this ability.  Do not change this once established or it will invalidate collected stats.
-		"AbilityBehavior"				"DOTA_ABILITY_BEHAVIOR_UNIT_TARGET"
-		"AbilityUnitTargetTeam" "DOTA_UNIT_TARGET_TEAM_FRIENDLY"
-		"AbilityUnitTargetType" "DOTA_UNIT_TARGET_HERO | DOTA_UNIT_TARGET_BASIC"
-		"SpellImmunityType"     "SPELL_IMMUNITY_ALLIES_YES"
+		"ID"							"7306"														// unique ID number for this ability.  Do not change this once established or it will invalidate collected stats.
+		"AbilityBehavior"				"DOTA_ABILITY_BEHAVIOR_PASSIVE"
+		"SpellImmunityType"				"SPELL_IMMUNITY_ALLIES_YES"
 		"FightRecapLevel"				"1"
-    "MaxLevel"              "6"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/chen_divine_favor.txt]: Undelete this :D

File: game/scripts/npc/abilities/slark_shadow_dance.txt [hunk 5]
```diff
@@ -65,7 +65,7 @@
       "07"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "scepter_aoe"                                     "325"
+        "scepter_aoe"                                     "400"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/slark_shadow_dance.txt]: Patch notes says 425. Valve is a pepega again?

File: game/scripts/npc/abilities/spectre_shadow_step.txt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		// General
+		//-------------------------------------------------------------------------------------------------------------
+		"ID"							"7851"														// unique ID number for this ability.  Do not change this once established or it will invalidate collected stats.
+		"AbilityBehavior"				"DOTA_ABILITY_BEHAVIOR_UNIT_TARGET | DOTA_ABILITY_BEHAVIOR_HIDDEN | DOTA_ABILITY_BEHAVIOR_SHOW_IN_GUIDES | DOTA_ABILITY_BEHAVIOR_NOT_LEARNABLE"
+		"AbilityUnitTargetTeam"			"DOTA_UNIT_TARGET_TEAM_ENEMY"
+		"AbilityUnitTargetType"			"DOTA_UNIT_TARGET_HERO"
+		"AbilityUnitTargetFlags"		"DOTA_UNIT_TARGET_FLAG_NOT_CREEP_HERO"
+		"SpellImmunityType"				"SPELL_IMMUNITY_ENEMIES_YES"
+		"FightRecapLevel"				"2"
+		"MaxLevel"						"3"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/spectre_shadow_step.txt]: MaxLevel should be 5.

File: game/scripts/npc/abilities/spectre_shadow_step.txt [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+		//-------------------------------------------------------------------------------------------------------------
+		"AbilityCooldown"				"70"
+
+		// Cost
+		//-------------------------------------------------------------------------------------------------------------
+		"AbilityManaCost"				"180"
+
+		// Special
+		//-------------------------------------------------------------------------------------------------------------
+		"AbilitySpecial"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/spectre_shadow_step.txt]: AbilitySpecial values should match Haunt probably.

File: game/scripts/npc/abilities/templar_assassin_trap_teleport.txt [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+			}
+			"02"
+			{
+				"var_type"					"FIELD_FLOAT"
+				"trap_duration"				"5.0"
+			}
+			"03"
+			{
+				"var_type"					"FIELD_INTEGER"
+				"trap_bonus_damage"			"250 300 350"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/templar_assassin_trap_teleport.txt]: This should have 5 values like Psionic Trap probably.

File: game/scripts/npc/abilities/tiny_tree_channel.txt [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+			}
+			"06"
+			{
+				"var_type"			"FIELD_FLOAT"
+				"interval"		"0.4"
+			}
+			"07"
+			{
+				"var_type"					"FIELD_INTEGER"
+				"bonus_damage"				"10 20 30 40"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/tiny_tree_channel.txt]: This should have 6 values. Same as Tree Grab probably.

File: game/scripts/npc/abilities/winter_wyvern_arctic_burn.txt [hunk 10]
```diff
@@ -41,7 +41,7 @@
       "03"
       {
         "var_type"                                        "FIELD_FLOAT"
-        "percent_damage"                                  "6 7 8 9 11 13"
+        "percent_damage"                                  "6 7 8 9 10 11"
```
Comment (chrisinajar) [inline game/scripts/npc/abilities/winter_wyvern_arctic_burn.txt]: lol, this has nothing to do with 7.22 :P

File: game/scripts/npc/abilities/windrunner_windrun.txt [hunk 11]
```diff
@@ -1,62 +1,85 @@
...
     // General
     //-------------------------------------------------------------------------------------------------------------
-    "ID"                                                  "5132"                            // unique ID number for this ability.  Do not change this once established or it will invalidate collected stats.
-    "AbilityBehavior"                                     "DOTA_ABILITY_BEHAVIOR_NO_TARGET | DOTA_ABILITY_BEHAVIOR_IMMEDIATE"
-    "SpellImmunityType"                                   "SPELL_IMMUNITY_ENEMIES_NO"
-    "SpellDispellableType"                                "SPELL_DISPELLABLE_YES"
+    "ID"          "5132"                            // unique ID number for this ability.  Do not change this once established or it will invalidate collected stats.
+    "AbilityBehavior"       "DOTA_ABILITY_BEHAVIOR_NO_TARGET | DOTA_ABILITY_BEHAVIOR_IMMEDIATE"
+    "SpellImmunityType"       "SPELL_IMMUNITY_ENEMIES_NO"
+    "SpellDispellableType"      "SPELL_DISPELLABLE_YES"
```
Comment (chrisinajar) [inline game/scripts/npc/abilities/windrunner_windrun.txt]: all you did was make this part way uglier, none of the values changed but the whitespace is awful now...

**Pull request merged**